libsignal-service-actix/src/push_service.rs

```diff
@@ -120,6 +106,20 @@ impl PushService for AwcPushService {
     type ByteStream = Box<dyn futures::io::AsyncRead + Unpin>;
     type WebSocket = AwcWebSocket;
 
+    fn new(
+        cfg: impl Into<ServiceConfiguration>,
+        credentials: Option<ServiceCredentials>,
+        user_agent: &'static str,
+    ) -> Self {
+        let cfg = cfg.into();
+        let client = get_client(&cfg, user_agent);
+        Self {
+            cfg,
+            credentials: credentials.and_then(|c| c.authorization()),
+            client,
+        }
+    }
+
```

Comment:
For the purposes of a clean diff, you may want to move this up a bit.

Reply:
Hmm, not sure what the best to achieve this is, since it's moved from `impl AwcPushService` to `impl PushService for AwcPushService`.

Reply:
Ack, not a problem.